### PR TITLE
Add KIM-1 support

### PR DESCRIPTION
**Pull request by rbergen**
This adds support for the MOS Technology KIM-1, in 4K and 60K RAM configurations.

The substance of the work in this PR was done by @davepl. As a collaborator on his fork, I _can_ make any changes that are necessary for the PR to be suitable for merging into the cc65 codebase.

### CODE REVIEW

File: include/kim1.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+/*                                 Hardware                                  */
+/*****************************************************************************/
+
+
+// Todo (davepl)
+//
+// #include <_6530.h>
+// #define RIOT3    (*(struct __6530*)0x1700)    // U25
+// #define RIOT2    (*(struct __6530*)0x1740)    // U28
+
```
Comment (mrdudz) [inline include/kim1.h:55]: Generally, please do not use C++ comments. That said, i'd prefer those lines to be either removed, or uncommented

**Comment by mrdudz** [comment]
Besides that one minor comment issue, it looks all fine to me - please fix and i will merge

File: include/kim1.h (hunk 1 above)
Comment (rbergen) [reply include/kim1.h]: I understand.

**Comment by mrdudz** [comment]
Cool, thanks!

**Comment by mrdudz** [comment]
Oops. Merged :) Cheers!

**Comment by pm100** [comment]
@rbergen @davepl I have written a 6502 debugger which works with sim65 emulator. I  would like to make it work with other emulators. I did a simple KIM emulator myself but wonder if there is a more 'mainstream' emulator that you used for your work on this PR

**Comment by rbergen** [comment]
@pm100 I can take no credit for the _actual_ work involved with what was merged here, I just handled the creation and follow-up of this PR - and yes, some minor code clean-ups related to that. 

The actual development and testing was done by @davepl. If he used an emulator I don't know though; Dave does own original KIM-1 hardware.

**Pull request merged**